pkg/crypto/crypto.go

```diff
@@ -61,14 +67,25 @@ func encrypt(plaintext string, key []byte) (string, error) {
 	writer.Close()
 	encoder.Close()
 
-	return string(msgBuffer.Bytes()), nil
+	// Remove unnecessary characters and trim new lines before storing as
+	// part of node Annotations username.
+	encryptedStr := string(msgBuffer.Bytes())
+	encryptedStr = strings.TrimPrefix(encryptedStr, startTag)
+	encryptedStr = strings.TrimSuffix(encryptedStr, endTag)
+	encryptedStr = strings.Trim(encryptedStr, "\n")
+	return encryptedStr, nil
 }
 
 // decrypt converts encrypted contents to plaintext using the given key as a passphrase
 func decrypt(encrypted string, key []byte) (string, error) {
 	if key == nil {
 		return "", errors.New("decryption passphrase cannot be empty")
 	}
+	// As we trim and remove unnecessary characters during encrypt call, we need to
+	// insert them back to encrypted string to get original value.
+	// 1. add start tag 2. add new lines 3.encrypted string 4. add new line 5. add end tag
+	// For more details : https://tools.ietf.org/id/draft-ietf-openpgp-rfc4880bis-06.html#cleartext-signature-framework
+	encrypted = startTag + newLine + newLine + encrypted + newLine + endTag
```

Comment:
In response to https://github.com/openshift/windows-machine-config-operator/pull/748#discussion_r728999370

For this code readability trumps maximum efficiency, this isn't used frequently so we don't need to prematurely optimize.
I don't have a strong opinion for either sprintf or + in this specific case, just pointing out that we should prioritize readability when we can.